### PR TITLE
Add GetR2Bucket

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
## Description

- Adds `GetR2Bucket` for getting info for individual buckets
- Changes CreationTime to *time.Time

## Has your change been tested?

All tests pass

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs. (soon™️)

### CODE REVIEW

File: r2_bucket.go [hunk 1]
```diff
@@ -40,6 +42,15 @@ type ListR2BucketsParams struct {
 	Cursor     string `url:"cursor,omitempty"`
 }
 
+type CreateR2BucketParameters struct {
+	Name string `json:"name,omitempty"`
```
Comment (sdnts) [inline r2_bucket.go]: FWIW you can also pass in a `locationHint` json parameter in here that accepts these values: https://developers.cloudflare.com/r2/buckets/data-location/#available-hints :D

Comment (Cyb3r-Jak3) [reply r2_bucket.go]: Good to know!

File: .changelog/1265.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
+r2_bucket: add support for getting a bucket
+```
+
+```release-note:breaking-change
+r2_bucket: change creation time from string to *time.Time
```
Comment (ferhatelmas) [inline .changelog/1265.txt:6]: why do we need to do this?

Comment (Cyb3r-Jak3) [reply .changelog/1265.txt:6]: I believe the standard is formatting time strings as `time.Time` so that users of the SDK are able to work with it easier.

Comment (ferhatelmas) [reply .changelog/1265.txt:6]: Definitely easier to use but was thinking if worth of a breaking change.

**Comment by Cyb3r-Jak3** [comment]
@jacobbednarz The docs are now live for this

**Comment by jacobbednarz** [comment]
thanks! i'll work with sid to get the docs corrected here as the structure is a little out but i see the intent.

**Pull request merged**